File: swift5/Sources/swift5/generating/singleton/MainSingleton.swift
import Foundation

fileprivate class Singleton {
    private static var instance: Singleton? = nil;
    private let value: Int = Int.random(in: 1..<Int.max)
    private init() {}
    public static func getInstance() -> Singleton {
        if (self.instance == nil) {
            let lock = NSLock()
            lock.lock()
            if (self.instance == nil) {
                self.instance = Singleton()
            }
            lock.unlock()
        }
        return instance!
    }
    public func getValue() -> Int {
        return self.value
    }
}

public class MainSingleton {
    public static func main() {
        let singleton1: Singleton = Singleton.getInstance();
        let singleton2: Singleton = Singleton.getInstance();
        print(singleton1.getValue())
        print(singleton2.getValue())
    }
}

File: python/generating/singleton/MainSingleton.py
from random import random
from threading import Lock

class Singleton:
    __instance = None

    def __init__(self):
        pass

    @classmethod
    def getInstance(cls):
        lock = Lock()
        if cls.__instance == None:
            lock.acquire()
            if cls.__instance == None:
                cls.__instance = Singleton()
                cls.__instance.__setValue(random())
            lock.release()
        return cls.__instance

    def __setValue(self, value):
        self.__value = value

    def getValue(self):
        return self.__value


print(Singleton().getInstance().getValue())
print(Singleton().getInstance().getValue())
print(Singleton().getInstance().getValue())